Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/MainActivity.java

```diff
@@ -113,8 +114,15 @@ private void analyze() {
     totalFrames =
         Integer.valueOf(mediaMetadataRetriever.extractMetadata(METADATA_KEY_VIDEO_FRAME_COUNT));
     // TODO: Add on demand frame read opposed to read all frames.
-    frameList = mediaMetadataRetriever.getFramesAtIndex(0, Math.max(0, totalFrames - SAFE_FRAMES));
+    for (int i = 0; i < totalFrames; i += FRAME_CHUNK_READ_SIZE) {
+      try{
+        frameList.addAll(mediaMetadataRetriever.getFramesAtIndex(
+            i, Math.min(totalFrames - i, FRAME_CHUNK_READ_SIZE)));
+      }catch (IllegalStateException e){
+        Log.d(ContentValues.TAG, "Video chunk read unsuccessful with: "+ e.getMessage());
```

Comment:
Shouldn't we abort the analysis?  Incomplete or completely missing frames will mess up the results won't they?

Reply:
@mwarton fixed. I added a command to kill the activity and fall back to previous activity to start the capture again.